src/Symfony/Component/VarDumper/Tests/Caster/FFICasterTest.php

```diff
@@ -0,0 +1,235 @@
+    }
+
+    /**
+     * @dataProvider scalarsDataProvider
+     */
+    public function testCastScalar(string $type, string $value, int $size, int $align)
+    {
+        $this->assertDumpEquals(<<<PHP
+        FFI\CData<$type> size $size align $align {
+          +cdata: $value
+        }
+        PHP, \FFI::new($type));
+    }
+
+    public function testVoidFunction()
+    {
+        $abi = $this->getDefaultAbi();
+
+        $this->assertDumpEquals(<<<PHP
+        $abi callable(): void {
+          returnType: FFI\CType<void> size 1 align 1 {}
+        }
+        PHP, \FFI::new('void (*)(void)'));
+    }
+
+    public function testIntFunction()
+    {
+        $abi = $this->getDefaultAbi();
+
+        $this->assertDumpEquals(<<<PHP
+        $abi callable(): uint64_t {
+          returnType: FFI\CType<uint64_t> size 8 align 8 {}
+        }
+        PHP, \FFI::new('unsigned long long (*)(void)'));
+    }
+
+    public function testFunctionWithArguments()
+    {
+        $abi = $this->getDefaultAbi();
+
+        $this->assertDumpEquals(<<<PHP
+        $abi callable(int32_t, char*): void {
+          returnType: FFI\CType<void> size 1 align 1 {}
+        }
+        PHP, \FFI::new('void (*)(int a, const char* b)'));
+    }
+
+    public function testCompositeStruct()
+    {
+        $ffi = \FFI::cdef(<<<'CPP'
+        typedef struct {
+            int x;
+            int y;
+        } Point;
+        typedef struct Example {
+            uint8_t a[32];
+            long b;
+            __extension__ union {
+                __extension__ struct {
+                    short c;
+                    long d;
+                };
+                struct {
+                    Point point;
+                    float e;
+                };
+            };
+            short f;
+            bool g;
+            int (*func)(
+                struct __sub *h
+            );
+        } Example;
+        CPP);
+
+        $var = $ffi->new('Example');
+        $var->func = (static fn (object $p) => 42);
+
+        $abi = $this->getDefaultAbi();
+        $longSize = $this->getLongSize();
+        $longType = 8 === $longSize ? 'int64_t' : 'int32_t';
+        $structSize = 56 + $longSize * 2;
+
+        $this->assertDumpEquals(<<<OUTPUT
+        FFI\CData<struct Example> size $structSize align 8 {
+          +a: FFI\CData<uint8_t[32]> size 32 align 1 {}
+          +$longType b: 0
+          +int16_t c: 0
+          +$longType d: 0
+          +point: FFI\CData<struct <anonymous>> size 8 align 4 {
+            +int32_t x: 0
+            +int32_t y: 0
+          }
+          +float e: 0.0
+          +int16_t f: 0
+          +bool g: false
+          +func: $abi callable(struct __sub*): int32_t {
+            returnType: FFI\CType<int32_t> size 4 align 4 {}
+          }
+        }
+        OUTPUT, $var);
+    }
+
+    private function getLongSize(): int
+    {
+        return \FFI::type('long')->getSize();
+    }
+
+    private function getDefaultAbi(): string
```

Comment:
let's inline this method and the previous one, ternary op to the rescue

Reply:
I inline a method for checking the length of a long depending on the OS, but I didn’t quite understand what you mean by "previous one, ternary op"?

Reply:
Ah, I understand! Done